src/OneExplorer/OneExplorer.ts

```diff
@@ -119,6 +120,26 @@ export class OneNode extends vscode.TreeItem {
 
     this.tooltip = `${this.node.path}`;
 
+    /**
+     * Returns custom view type for each nodes
+     */
+    const getCustomViewType = (node: Node): string|undefined => {
+      if (node.type === NodeType.config) {
+        return CfgEditorPanel.viewType;
+      } else if (node.type === NodeType.derivedModel && node.ext === '.circle') {
+        return CircleViewerProvider.viewType;
+      } else {
+        return undefined;
+      }
+    };
+
+    this.command = getCustomViewType(node) ? {
+      command: 'vscode.openWith',
+      title: 'Open File',
+      arguments: [node.uri, getCustomViewType(node)]
+    } :
+                                             undefined;
```

Comment:
This line align is done by clang-format... :frowning_face:

Reply:
Maybe I can take the replacement, I think clang-format on/off makes it rather complicated.